lib/ansible/modules/network/nxos/nxos_vpc_interface.py

```diff
@@ -262,9 +262,10 @@ def main():
     active_peer_link = None
 
     if portchannel not in get_portchannel_list(module):
```

Comment:
Do we still need this check?

Reply:
This check is for the normal case where portchannel is a string. I could remove it and leave the single check on line 265.

Reply:
@rahushen I think we can ignore this check because portchannel is actually digit, it is just we have arg param as str. 
The added check already does the typecasting `int(portchannel)`. So the first check looks unnecessary to me.

Reply:
@trishnaguha This has been taken care of now! Also, this needs to be merged into 2.4.1 as well.

Reply:
@trishnaguha ... thinking more about this, I think the check is needed. 

get_portchannel_list(module) returns the portchannel list from device (string on most devices and int on only N1). 

`if not portchannel.isdigit() or int(portchannel) not in get_portchannel_list(module): `

The above check will fail for all other platforms because type casting portchannel to int will break string equality check. Hence we need both checks.

Reply:
Ok. Merging this on devel and cherry-picking to 2.4.